packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -791,45 +792,47 @@ export default function makeKernelKeeper(
     return kpid;
   }
 
+  /**
+   * @param {string} kernelSlot
+   * @returns {PromiseRecord}
+   */
   function getKernelPromise(kernelSlot) {
     insistKernelType('promise', kernelSlot);
-    const p = { state: getRequired(`${kernelSlot}.state`) };
-    switch (p.state) {
+    const state = getRequired(`${kernelSlot}.state`);
+    const refCount = Number(kvStore.get(`${kernelSlot}.refCount`));
+    switch (state) {
       case undefined: {
         throw Fail`unknown kernelPromise '${kernelSlot}'`;
       }
       case 'unresolved': {
-        p.refCount = Number(kvStore.get(`${kernelSlot}.refCount`));
-        p.decider = kvStore.get(`${kernelSlot}.decider`);
-        if (p.decider === '') {
-          p.decider = undefined;
+        let decider = kvStore.get(`${kernelSlot}.decider`);
+        if (decider === '') {
+          decider = undefined;
         }
-        p.policy = kvStore.get(`${kernelSlot}.policy`) || 'ignore';
-        // @ts-expect-error get() may fail
-        p.subscribers = commaSplit(kvStore.get(`${kernelSlot}.subscribers`));
-        p.queue = Array.from(
+        const policy = kvStore.get(`${kernelSlot}.policy`) || 'ignore';
+        const subscribers = commaSplit(
+          getRequired(`${kernelSlot}.subscribers`),
```

Comment:
Is subscribers guaranteed to be defined? Any hurt to support it not being?

```suggestion
          kvStore.get(`${kernelSlot}.subscribers`) || '',
```

Reply:
if `state` is `unresolved`, then `.subscribers` is supposed to be defined, and it would be some sort of bug if it were missing, but yeah in the spirit of being maximally tolerant (or enabling some future optimization where we omit the entry until there is at least one subscriber, which isn't a bad idea TBH) it'd be fine to do it that way.. I'll make the change